cdk/domino_cdk/util.py

```diff
@@ -146,3 +153,60 @@ def overlay(alpha: dict, omega: dict) -> dict:
             return {}
         base_dict = check_type(dictionaries[0])
         return base_dict if len(dictionaries) == 1 else overlay(base_dict, cls.deep_merge(*dictionaries[1:]))
+
+    @staticmethod
+    def create_lambda(
+        scope: cdk.Construct,
+        stack_name: str,
+        dirname: str,
+        name: str,
+        environment: Dict[str, str],
+        resources: List[str],
+        actions: List[str],
+    ) -> cdk.Construct:
+        with open(path.join(dirname, "lambda", name + ".py"), encoding="utf-8") as fp:
+            on_event_code_body = fp.read()
+        on_event = lambda_.Function(
+            scope,
+            name + "_on_event",
+            function_name=f"{stack_name}-{name}",
+            runtime=lambda_.Runtime.PYTHON_3_7,
+            handler="index.on_event",
+            code=lambda_.InlineCode(on_event_code_body),
+            environment=environment,
+            timeout=cdk.Duration.seconds(600),  # default is 3 seconds
+            log_retention=logs.RetentionDays.ONE_DAY,  # defaults to never delete logs
+        )
+        statement = iam.PolicyStatement()
+        for r in resources:
+            statement.add_resources(r)
+        for a in actions:
+            statement.add_actions(a)
+        on_event.add_to_role_policy(statement)
+
+        is_complete = None
+        try:
+            with open(path.join(dirname, "lambda", name + "_is_complete.py"), encoding="utf-8") as fp:
+                is_complete_code_body = fp.read()
+            is_complete = lambda_.Function(
+                scope,
+                name + "_is_complete",
+                runtime=lambda_.Runtime.PYTHON_3_7,
+                handler="index.is_complete",
+                code=lambda_.InlineCode(is_complete_code_body),
+                # timeout defaults to 2 minutes
+                log_retention=logs.RetentionDays.ONE_DAY,  # defaults to never delete logs
+            )
+            is_complete.add_to_role_policy(statement)
+        except IOError:
+            pass
```

Comment:
What would IOError in this? A missing file?

Reply:
I'm still living in Python < 3.3